src/framework/audio/internal/rpc/rpctypes.h

```diff
@@ -0,0 +1,144 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2021 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_RPCTYPES_H
+#define MU_AUDIO_RPCTYPES_H
+
+#include <string>
+#include <map>
+#include <memory>
+
+namespace mu::audio::rpc {
+enum class TargetName {
+    Undefined = 0,
+    Sequencer = 1
+};
+
+struct Target {
+    TargetName name = TargetName::Undefined;
+    int instanceID = 0;
+    Target() = default;
+    Target(TargetName n)
+        : name(n) {}
+
+    inline bool operator ==(const Target& other) const { return other.name == name && other.instanceID == instanceID; }
+    inline bool operator !=(const Target& other) const { return !this->operator ==(other); }
+};
+
+using Method = std::string;
+
+//! NOTE Depending on the implementation of the RPC channel,
+//! the data may or may not need to be serialized.
+//! For example, if we communicate with Web Worker or a separate process,
+//! then in this case it is impossible to transfer objects, structures, pointers and etc - the data must be serialized.
+//! If we communicate between threads in the same application, then the data does not need to be serialized.
+
+class Args
```

Comment:
it looks like ActionData :)
can we combine and put it in the global module?

Reply:
I'm not sure, yes, the current implementation same, but these are different types and may be changed by new requirements.